File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/path/OnlyTurnPoint.java
package org.firstinspires.ftc.teamcode.control.path;

import org.firstinspires.ftc.teamcode.control.system.Functions;

public class OnlyTurnPoint extends LockedPathPoint {
    public double dh;
    public OnlyTurnPoint(String signature, double x, double y, double followDistance, double h, double dh) {
        super(signature, x, y, followDistance, h);
        this.dh = dh;
    }

    public OnlyTurnPoint(String signature, double x, double y, double followDistance, double h, double dh, Functions.Function func) {
        super(signature, x, y, followDistance, h, func);
        this.dh = dh;
    }

    public OnlyTurnPoint(OnlyTurnPoint p) {
        this(p.signature, p.x, p.y, p.followDistance, p.h, p.dh, p.func);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/path/OnlyTurnPoint.kt
package org.firstinspires.ftc.teamcode.control.path

import org.firstinspires.ftc.teamcode.control.system.Functions
import org.firstinspires.ftc.teamcode.util.Angle

class OnlyTurnPoint(
    signature: String,
    x: Double,
    y: Double,
    followDistance: Double,
    h: Angle,
    var dh: Angle,
    func: Functions.Function? = null
) : LockedPathPoint(signature, x, y, followDistance, h, func) {

    override val copy: PathPoint get() = OnlyTurnPoint(signature, x, y, followDistance, h, dh, func)
    override fun toString(): String {
        return String.format(
            "%s, %.1f, %.1f, %.1f, %.1f, %.1f, %s",
            signature,
            x,
            y,
            followDistance,
            h,
            dh,
            func
        )
    }
}
